Indicators/ArimaIndicator.cs

```diff
@@ -0,0 +1,307 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MathNet.Numerics;
+using MathNet.Numerics.LinearAlgebra.Double;
+using MathNet.Numerics.LinearRegression;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// An ARIMA is a time series model which can be used to describe a set of data. In particular,with Xₜ
+    /// representing the series, the model assumes the data are of form (after differencing <see cref="_d" /> times):
+    /// <para>
+    ///     Xₜ = c + εₜ + ΣᵢφᵢXₜ₋ᵢ +  Σᵢθᵢεₜ₋ᵢ
+    /// </para>
+    /// where the first sum has an upper limit of <see cref="_p" /> and the second <see cref="_q" />.
+    /// </summary>
+    public class ArimaIndicator : TimeSeriesIndicator, IIndicatorWarmUpPeriodProvider
+
+    {
+        /// <summary>
+        /// Differencing coefficient. Determines how many times the series should be differenced before fitting the
+        /// model.
+        /// </summary>
+        private readonly int _d;
+
+        private readonly bool _intercept;
+
+        /// <summary>
+        /// AR coefficient.
+        /// </summary>
+        private readonly int _p;
+
+        /// <summary>
+        /// MA Coefficient.
+        /// </summary>
+        private readonly int _q;
+
+        private readonly RollingWindow<double> _rollingData;
+
+        private List<double> _residuals;
+
+        /// <summary>
+        /// A dictionary, indexed by "AR" and "MA", containing their respective, fitted parameters.
+        /// </summary>
+        public Dictionary<string, double[]> Parameters;
```

Comment:
Are these going to be consumed by a user of the indicator?
Should they directly be 3 properties of the indicator AR/MA & intercept?

Reply:
These should be accessible imo in case one wishes to implement the fitted ARIMA parameters for some other common purposes other than forecasting (e.g., starting parameters for fitting a SARIMA).